File: placementprepration/graph/BFS.cpp
#include<iostream>
#include<vector>
#include<queue>
using namespace std;
class solution {
    public:
    vector<int>BFS(vector<vector<int>>adj){
       int v=adj.size();
       int vis[v]={0};
       vis[0]=1;
       queue<int>q;
       vector<int>res;
       q.push(0);
       while(!q.empty()){
            int node=q.front();
            q.pop();
            res.push_back(node);
            for(auto it : adj[node]){
                if(!vis[it]){
                    vis[it]=1;
                    q.push(it);
                }
            }
        }
       return res;
    }
    void addEdge(vector<int> adj[], int u, int v){
        adj[u].push_back(v);
        adj[v].push_back(u);
    }
    void printans(vector<int>&ans){
        for(int i=0;i<ans.size();i++){
            cout<<ans[i]<<" ";
        }
    }

};
int main(){
    vector<int>adj[6];
    

}

File: placementprepration/graph/BFS.py
import sys
from collections import deque


def bfs(adj):
    v = len(adj)
    vis = [0] * v
    vis[0] = 1
    q = deque([0])
    res = []
    while q:
        node = q.popleft()
        res.append(node)
        for nei in adj[node]:
            if not vis[nei]:
                vis[nei] = 1
                q.append(nei)
    return res


def main():
    data = list(map(int, sys.stdin.read().strip().split()))
    if len(data) < 2:
        return
    it = iter(data)
    n = int(next(it))
    m = int(next(it))
    adj = [[] for _ in range(n)]
    for _ in range(m):
        u = int(next(it))
        v = int(next(it))
        adj[u].append(v)
        adj[v].append(u)
    res = bfs(adj)
    print(" ".join(map(str, res)))


if __name__ == "__main__":
    main()
